haystack/telemetry/_telemetry.py

```diff
@@ -144,18 +144,17 @@ def pipeline_running(pipeline: "Pipeline") -> Optional[Tuple[str, Dict[str, Any]
     pipeline._last_telemetry_sent = datetime.datetime.now()
 
     # Collect info about components
-    pipeline_description = pipeline.to_dict()
     components: Dict[str, List[Dict[str, Any]]] = defaultdict(list)
-    for component_name, component in pipeline_description["components"].items():
-        instance = pipeline.get_component(component_name)
+    for component_name, instance in pipeline.graph.nodes(data="instance"):
+        component_qualified_class_name = ".".join([type(instance).__module__, type(instance).__name__])
```

Comment:
We shouldn't attempt to resolve the namespace of the component here - The serialization module should expose a function to generate this, which is then also used by the default `from/to_dict` implementations.